Magefile.go

```diff
@@ -0,0 +1,127 @@
+//+build mage
+
+package main
+
+import (
+	"archive/zip"
+	"encoding/json"
+	"fmt"
+	"github.com/magefile/mage/mg"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	// mage:import
+	"github.com/grafana/grafana-plugin-sdk-go/build"
+	"github.com/grafana/grafana-plugin-sdk-go/build/utils"
+)
+
+func CopyArtifacts() error {
+	// TODO: Only try to remove if dist exists, check error
+	_ = os.RemoveAll("dist")
+	if err := os.MkdirAll("dist", 0775); err != nil {
+		return err
+	}
+
+	fpaths := []string{"src/plugin.json", "README.md"}
+	fis, err := ioutil.ReadDir("img")
+	if err != nil {
+		return err
+	}
+	for _, fi := range fis {
+		fpaths = append(fpaths, path.Join("img", fi.Name()))
+	}
+
+	for _, fpath := range fpaths {
+		if err := utils.CopyFile(fpath, path.Join("dist", path.Base(fpath))); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readPluginJSON() (map[string]interface{}, error) {
+	byteValue, err := ioutil.ReadFile(path.Join("src", "plugin.json"))
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal([]byte(byteValue), &result)
+	return result, err
+}
+
+// Zip builds the plugin zip archive.
+func Zip() error {
+	mg.Deps(CopyArtifacts, build.BuildAll)
```

Comment:
```suggestion
func ZipDist() error {
```

This would be more useful if it could get called after other things have managed the `dist` folder